src/components/DateRangePopup.tsx

```diff
@@ -135,6 +138,16 @@ const DateRangePopup: React.FC<CustomSelectModalProps> = ({
     }
   };
 
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.localStorage) {
+      const storedDates = localStorage.getItem('selectedRangeArray');
+      if (storedDates) {
+        const dateArray = JSON.parse(storedDates);
```

Comment:
@Rushikesh-Sonawane99 add try catch block to avoid json.parse error

Reply:
@itsvick Added  try catch block